indexd/index/drivers/alchemy.py

```diff
@@ -346,6 +347,20 @@ def ids(self,
                         ))
                     query = query.filter(IndexRecord.did.in_(sub.subquery()))
 
+            if urls_metadata is not None and urls_metadata:
+                query = query.join(IndexRecord.urls).join(IndexRecordUrl.url_metadata)
+                for url_key, url_dict in urls_metadata.items():
+                    query = query.filter(IndexRecordUrlMetadata.url.contains(url_key))
```

Comment:
Contain would check if url contains a substring that matches the passed url. No wildcard is allowed in this case. We can also use SQL LIKE and use % as wildcard.

Reply:
Not validating urls_metadata schema here. If client pass in valid json string with invalid schema, 500 will be thrown. IndexClient should handle this?